### DISCUSSION TITLE
Build level support

### DISCUSSION THREAD

**Discussion by ghost** [Sketchware-Pro/Sketchware-Pro#601] [Q&A] [answered]
Why do you always delay to update the custom gradle build.The app is totally nice (Rc04) with few errors but why do you force us to use libraries that we don't want❔
.
Add it up please😒

**Comment by j-px** [comment]
Sketchware Pro doesn't use Gradle. The build.gradle files you find in /Internal storage/.sketchware/mysc/... are only compatibility with Android Studio when exporting projects.

**Reply by ghost** [reply]
I agree....but what makes it not be an added feature❔....it would be helpful

**Reply by j-px** [reply]
It's not likely soon, probably never if I'm honest. Gradle is not designed to work on Android.

**Reply by ghost** [reply]
Then why does AIDE SDK support it even if it's lightweight by 36Mibs❔🤔

**Reply by j-px** [reply]
It probably uses an old Gradle version, or not even Gradle at all. Anyway, Sketchware Pro and AIDE have major differences, and comparing them is kinda like comparing apples and bananas.

**Reply by ghost** [reply]
😂maybe.......
Let us then edit the automatic compile cript permission then❔

**Reply by j-px** [reply]
There's no such script, what Sketchware Pro does to compile projects is code and can't be changed easily like a script. You'd have to re-compile the whole app.

**Reply by ghost** [reply]
Make it a beta so we can test on it....
Custom java points as strings

**Reply by j-px** [reply]
Test what? Even if, porting Gradle to Android to work will take a huge amount of effort. Just look at CodeAssist and what it's trying to do currently. Still unstable.

**Reply by ghost** [reply]
You are not straight with your answers

**Reply by j-px** [reply]
what?

**Reply by ghost** [reply]
Why does appcomapt switch activate gradle and it's properties and still deny us to edit gradle❔
making sense right

**Reply by j-px** [reply]
You didn't get me right. Enabling AppCompat does **not** enable Gradle. Gradle can't be enabled at all.

**Reply by ghost** [reply]
I don't get it......
why does it enable all the android x libraries ❔🤔

**Reply by j-px** [reply]
Ask Sketchware developers. We're just keeping compatibility.

**Reply by ghost** [reply]
plugins {
	id 'com.android.application'
}

android {
	compileSdkVersion 28
	
	useLibrary 'org.apache.http.legacy'
	
	defaultConfig {
		applicationId "com.MisterApps.Sheetdb"
		minSdkVersion 21
		targetSdkVersion 28
		versionCode 1
		versionName "1.0"
	}
	
	buildTypes {
		release {
			minifyEnabled false
			proguardFiles getDefaultProguardFile('proguard-android-optimize.txt'), 'proguard-rules.pro'
		}
	}
}

dependencies {
	implementation fileTree(dir: 'libs', include: ['*.jar'])

//📌WHY DOES CODE APPEAR WHEN APP COMPAT IS ENABLED//

	implementation 'com.google.code.gson:gson:2.8.7'

//RECYCLER VIEW//
//GSON //
//FRAGMENT ADAPTER//

	implementation 'com.squareup.okhttp3:okhttp:3.9.1'
}


Question is now straight 😂

**Comment by ghost** [comment, marked as answer]
The Automatic Build.gradle file needs to be made automatic for smoother experience on knowledgeable coders ✔️📌
